app/scripts/metamask-controller.js

```diff
@@ -468,7 +468,7 @@ module.exports = class MetamaskController extends EventEmitter {
       whitelistPhishingDomain: this.whitelistPhishingDomain.bind(this),
 
       // AddressController
-      setAddressBook: this.addressBookController.set.bind(this.addressBookController),
+      setAddressBook: nodeify(this.addressBookController.set, this.addressBookController),
```

Comment:
Why does this need to be converted to use a callback? Is this just to follow the convention set by the rest of the background API, or is this assumed to take a callback somewhere?

Reply:
Yes, this is largely to follow the convention more closely but also to ensure that the return value isn't discarded since it's used in the action (handwaving away some details, dnode doesn't work without a callback 👋 👋).

Reply:
Ah right, dnode. That's a very good reason.